src/main/java/seedu/duke/operationlist/PassengerList.java

```diff
@@ -0,0 +1,453 @@
+        }
+    }
+
+    private void getDepartureDate() {
+        departureDate = PassengerInfo.getDepartureDate();
+    }
+
+    private void getDepartureTime(String passengerDetail) throws SkyControlException {
+        if (isAdd) {
+            departureTime = getSubstringBetweenDelimiters(passengerDetail,
+                    DEPARTURE_TIME_DELIMITER, FLIGHT_NUMBER_DELIMITER);
+        } else if (isDelete) {
+            startIndex = passengerDetail.indexOf(DEPARTURE_TIME_DELIMITER)
+                    + DEPARTURE_TIME_DELIMITER.length();
+            assert startIndex >= 0;
+            departureTime = passengerDetail.substring(startIndex);
+            departureTime = departureTime.toUpperCase();
+        }
+    }
+
+    static void getFlightNumber(String passengerDetail) throws SkyControlException {
+        if (isAdd) {
+            flightNumber = getSubstringBetweenDelimiters(passengerDetail,
+                    FLIGHT_NUMBER_DELIMITER, GATE_NUMBER_DELIMITER);
+        } else if (isDelete) {
+            flightNumber = getSubstringBetweenDelimiters(passengerDetail,
+                    FLIGHT_NUMBER_DELIMITER, SEAT_NUMBER_DELIMITER);
+        }
+    }
+
+    static String getFlightNumberForSync(String passengerDetail) {
+        try {
+            getFlightNumber(passengerDetail);
+        } catch (SkyControlException e) {
+            ui.showError(e.getMessage());
+        }
+        return flightNumber;
+    }
+
+    private void getGateNumber(String passengerDetail) throws SkyControlException {
+        gateNumber = getSubstringBetweenDelimiters(passengerDetail,
+                GATE_NUMBER_DELIMITER, BOARDING_GROUP_DELIMITER);
+    }
+
+    private void getBoardingGroup(String passengerDetail) throws SkyControlException {
+        String boardingGroupDetail = getSubstringBetweenDelimiters(passengerDetail,
+                BOARDING_GROUP_DELIMITER, SEAT_NUMBER_DELIMITER);
+        try {
+            boardingGroup = Integer.parseInt(boardingGroupDetail);
+        } catch (Exception e) {
+            throw new SkyControlException(ui.getErrorMessage());
+        }
+        assert boardingGroup >= 0 : "Invalid boarding group";
+    }
+
+    private void getSeatNumber(String passengerDetail) throws SkyControlException {
+        if (isAdd) {
+            seatNumber = getSubstringBetweenDelimiters(passengerDetail,
+                    SEAT_NUMBER_DELIMITER, BOARDING_TIME_DELIMITER);
+        } else if (isDelete) {
+            seatNumber = getSubstringBetweenDelimiters(passengerDetail,
+                    SEAT_NUMBER_DELIMITER, DEPARTURE_TIME_DELIMITER);
+        }
+    }
+
+    private void getBoardingTime(String passengerDetail) throws SkyControlException {
+        int indexOfBoardingTime = passengerDetail.indexOf(BOARDING_TIME_DELIMITER);
+        int indexOfSeatNumber = passengerDetail.indexOf(SEAT_NUMBER_DELIMITER);
+        if (indexOfBoardingTime < 0 || indexOfBoardingTime < indexOfSeatNumber) {
```

Comment:
This is a complicated expression. Perhaps you can consider putting the check in another method and return a boolean value instead.